### PR TITLE
Allow aligning objects bigger than the cache line size

### PR DESCRIPTION
**Pull request by msimberg**

### CODE REVIEW

File: hpx/util/cache_aligned_data.hpp [hunk 1]
```diff
@@ -30,7 +30,8 @@ namespace hpx { namespace util
 
         // pad to cache line size bytes
         Data data_;
-        char cacheline_pad[threads::get_cache_line_size() - sizeof(Data)];
+        char cacheline_pad[threads::get_cache_line_size() -
+            (sizeof(Data) % threads::get_cache_line_size())];
     };
```
Comment (hkaiser) [inline hpx/util/cache_aligned_data.hpp]: We'll need to adapt the `static_assert` above in this case.

Comment (hkaiser) [inline hpx/util/cache_aligned_data.hpp]: We'll need to adapt the `static_assert` above in this case.

Comment (msimberg) [reply hpx/util/cache_aligned_data.hpp]: Indeed, good catch. Thanks!

Comment (msimberg) [reply hpx/util/cache_aligned_data.hpp]: Indeed, good catch. Thanks!

**Comment by msimberg** [comment]
The codacy warning is on point but a must in this case.

**Comment by hkaiser** [comment]
> The codacy warning is on point but something we can't really change in this case...

I have flagged the issue to be ignored. It will not show up for future changes to this PR anymore.

**Comment by msimberg** [comment]
This should be good to go now.

**Pull request merged**